File: components/string-utilities/src/main/java/org/cloudfoundry/credhub/utils/ResourceReader.java
package org.cloudfoundry.credhub.utils;

import java.io.IOException;
import java.net.URL;

import org.springframework.stereotype.Component;

import com.google.common.io.Resources;

@Component
public class ResourceReader {
  public String readFileToString(final String fileName) throws IOException {
    final URL resource = Resources.getResource(fileName);
    return Resources.toString(resource, StringUtil.UTF_8);
  }
}

File: components/string-utilities/src/main/kotlin/org/cloudfoundry/credhub/utils/ResourceReader.kt
package org.cloudfoundry.credhub.utils

import com.google.common.io.Resources
import org.springframework.stereotype.Component
import kotlin.text.Charsets.UTF_8

@Component
class ResourceReader {
    fun readFileToString(fileName: String): String {
        val resource = Resources.getResource(fileName)
        return Resources.toString(resource, UTF_8)
    }
}
